build.gradle

```diff
@@ -61,16 +69,46 @@ repositories {
     mavenCentral()
 }
 
+// Write the plugin's classpath to a file to share with the tests
+task createClasspathManifest {
+  def outputDir = file("$buildDir/$name")
+
+  inputs.files sourceSets.main.runtimeClasspath
+  outputs.dir outputDir
+
+  doLast {
+    outputDir.mkdirs()
+    file("$outputDir/plugin-classpath.txt").text = sourceSets.main.runtimeClasspath.join("\n")
+  }
+}
+
+tasks.test.dependsOn tasks.createClasspathManifest
+if (findProject(':vendor') != null) {
+  def vendorModelRegistry = project.evaluationDependsOn(':vendor').getModelRegistry()
```

Comment:
Since I'm using the new TestKit from Gradle and I require the protoc and protobuf library in the test, I need them to be built before I run the test. This if-statement ensure these requirements are meet for the test task which execute the TestKit test. I force the model to "realize" (aka instantiate) the required node from the project 'vendor'. I can then get the binaries instances and add them as a dependency of the test task. When the test task is executed, the protoc and protobuf library will get build before.

I'm aware the ModelRegistry is a internal feature and I'm actively searching for an alternative in the public API. I'm hoping to find a better solution in my next PR.